apbs/create-secret-and-configmap-during-provision.md

```diff
@@ -0,0 +1,44 @@
+# Abstract
+
+This document outlines a proposal to create an additional configmap for services accessible via a mobile client, this configmap would compliment the secret we currently create during the provision of each service via an APB 
+
+## Terms
+
+- Mobile Client (a mobile device in the public domain running a developers application)
+- APB (ansible playbook bundle)
+
+## Problem Description
+
+Our current approach is to create a secret during the provision of a new mobile service. This secret contains all of the information required to allow the UI to show the service and how to access the service, it also includes any information needed to create the mobile client configuration. This means we expose too much information and also need to filter out any sensitive information (usernames password etc) when constructing our mobile configuration to avoid it being exposed to the client and ending up in the public domain. As each service is different, it often requires a custom piece of code to filter out the fields that we don't want or need exposed. 
```

Comment:
The 'Service Instance' created by the catalog is probably a better thing to determine if the service should be shown in the UI.
This is already the case on the Overview screen in the 'Provisioned Services' section.
Mobile services are likely to show up in that section too, but have some additional UI when you expand it

Reply:
Sure however the service instance wont have the credentials and additional info right? So the UI will still need to read the secret?

Reply:
Why does the UI need to read the secret?

Reply:
I may be out of the loop on the UX but originally we had been showing a screen where the url and login credentials were shown to allow a developer to jump into the service?

Reply:
A good thing to discuss with UX as it hasn't been discussed yet. Just the idea of a link i.e. 'Dashbaord URL' showing up was discussed
cc @cshinn.

Either way, reading the secret would/should only be needed for that (if the user has perms to read the secret).
Reading the Service Instance to populate the 'Provisioned Services' section (as it does now) make sense to me

Reply:
so maybe a separate proposal to replace using the secret with using the ServiceInstance? Either way it is not a  major part of this proposal

Reply:
I'm good with a separate proposal for that